src/ImageSharp/Processing/Transforms/Transform.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Six Labors and contributors.
+// Licensed under the Apache License, Version 2.0.
+
+using System.Numerics;
+using SixLabors.ImageSharp.PixelFormats;
+using SixLabors.ImageSharp.Processing;
+using SixLabors.ImageSharp.Processing.Processors;
+
+namespace SixLabors.ImageSharp
+{
+    /// <summary>
+    /// Extension methods for the <see cref="Image{TPixel}"/> type.
+    /// </summary>
+    public static partial class ImageExtensions
+    {
+        /// <summary>
+        /// Transforms an image by the given matrix.
+        /// </summary>
+        /// <typeparam name="TPixel">The pixel format.</typeparam>
+        /// <param name="source">The image to skew.</param>
+        /// <param name="matrix">The transformation matrix.</param>
+        /// <returns>The <see cref="Image{TPixel}"/></returns>
+        public static IImageProcessingContext<TPixel> Transform<TPixel>(this IImageProcessingContext<TPixel> source, Matrix3x2 matrix)
```

Comment:
An API idea:
Maybe we could provide a few shortcuts for common affine transforms:
```C#
public enum TransformOrigin { Center, TopLeft }

public static IImageProcessingContext<TPixel> Transform<TPixel>(this IImageProcessingContext<TPixel> source, float rotationAngle, Vector2 scale, Vector2 translation, TransformOrigin origin = TransformOrigin.Center) { ... }

public static IImageProcessingContext<TPixel> Transform<TPixel>(this IImageProcessingContext<TPixel> source, float rotationAngle = 0, float uniformScale = 1, float translationX = 0, float translationY = 0, TransformOrigin origin = TransformOrigin.Center) { ... }

// etc ...

```

Reply:
Aye maybe. Let's get the basics in first then expand. Those `Transform.Origin` values could maybe complicate things